cluster_management/src/main/java/com/pinterest/rocksplicator/Participant.java

```diff
@@ -218,26 +235,31 @@ public static void main(String[] args) throws Exception {
     final String zkEventHistoryStr = cmd.getOptionValue(handoffEventHistoryzkSvr, "");
     final String resourceConfigPath = cmd.getOptionValue(handoffEventHistoryConfigPath, "");
     final String resourceConfigType = cmd.getOptionValue(handoffEventHistoryConfigType, "");
+    final String shardMapPath = cmd.getOptionValue(handoffClientEventHistoryJsonShardMapPath, "");
 
     /**
      * Note that the last parameter is empty, since we don't dictate
      * maxEventsToKeep from participants.
      */
-    staticParticipantLeaderEventsLogger =
-        new LeaderEventsLoggerImpl(instanceName,
-            zkEventHistoryStr, clusterName, resourceConfigPath, resourceConfigType, Optional.empty());
+    staticParticipantLeaderEventsLogger = new LeaderEventsLoggerImpl(instanceName,
+        zkEventHistoryStr, clusterName, resourceConfigPath, resourceConfigType, Optional.empty());
 
     if (runSpectator) {
-      staticSpectatorLeaderEventsLogger =
-          new LeaderEventsLoggerImpl(instanceName,
-              zkEventHistoryStr, clusterName, resourceConfigPath, resourceConfigType,
-              Optional.of(128));
+      staticSpectatorLeaderEventsLogger = new LeaderEventsLoggerImpl(instanceName,
+          zkEventHistoryStr, clusterName, resourceConfigPath, resourceConfigType, Optional.of(128));
+
+      if (shardMapPath != null && !shardMapPath.isEmpty()) {
+        staticClientLeaderEventsLogger = new LeaderEventsLoggerImpl(instanceName,
+            zkEventHistoryStr, clusterName, resourceConfigPath, resourceConfigType, Optional.empty());
+
+        staticClientShardMapLeaderEventLoggerDriver = new ClientShardMapLeaderEventLoggerDriver(
```

Comment:
can u remind us what is the "Driver" vs the EventsLogger?

Reply:
Driver here is driving the notification on changes in the file. LeaderEventsLogger is the main class common in all cases. We use special implementation that encapsulates LeaderEventsLogger for specific needs. E.g.. ClientShardMapLeaderEventsLogger will process ShardMap and convert it into interest events to log for LeaderEventsLogger. Similarly in case of Spectator, we use ExternalViewBasedLeaderEventsLogger that converts the ExternalView based notifications into interesting events. ExternalView is being driven by helix notifications. We don't have one for local file watching... so ClientShardMapLeaderEventLoggerDriver basically does that part of driving notifications from changes in shard_map file and forwarding it down the line.